src/tiledb/cloud/soma.py

```diff
@@ -0,0 +1,143 @@
+import logging
+
+import tiledb
+from tiledb.cloud import dag
+from tiledb.cloud._common import functions
+
+from typing import Dict, Optional
+
+_DEFAULT_RESOURCES = {"cpu": "8", "memory": "8Gi"}
+"""Default resource size; equivalent to a "large" UDF container."""
+
+
+def ingest_h5ad(
+    *,
+    output_uri: str,
+    input_uri: str,
+    measurement_name: str,
+    extra_tiledb_config: Optional[Dict[str, object]],
+    platform_config: Optional[Dict[str, object]],
+    ingest_mode: str,
+) -> None:
+    """Performs the actual work of ingesting H5AD data into TileDB.
+
+    :param output_uri: The output URI to write to. This will probably look like
+        ``tiledb://namespace/some://storage/uri``.
+    :param input_uri: The URI of the H5AD file to read from. This file is read
```

Comment:
Empirically I've found that:

* If we use `anndata.read_h5ad(foo)` where `foo` is a `tiledb.VFS`, this does work
* If we use `anndata.read_h5ad(foo, "r")` then this does _not_ work
  * Note that `"r"` is important for memory reduction in memory-constrained environments (e.g. UDF nodes)
  * Issue is that `anndata.read_h5ad` docstrings say the first argument is a path, but in reality, a `tiledb.VFS` object plumbs almost all the way through ...
  * ... except with `"r"` mode where at one point in the anndata code it does assume the first arg is a path per se